docs/ops/sort/NonMaxSuppression_5.md

```diff
@@ -0,0 +1,121 @@
+## NonMaxSuppression<a name="NonMaxSuppression"></a> {#openvino_docs_ops_sort_NonMaxSuppression_5}
+
+**Versioned name**: *NonMaxSuppression-5*
+
+**Category**: *Sorting and maximization*
+
+**Short description**: *NonMaxSuppression* performs non maximum suppression of the boxes with predicted scores.
+
+**Detailed description**: *NonMaxSuppression* performs non maximum suppression algorithm as described below:
+
+1.  Let `B = [b_0,...,b_n]` be the list of initial detection boxes, `S = [s_0,...,s_N]` be  the list of corresponding scores.
+2.  Let `D = []` be an initial collection of resulting boxes.
+3.  If `B` is empty then go to step 8.
+4.  Take the box with highest score. Suppose that it is the box `b` with the score `s`.
+5.  Delete `b` from `B`.
+6.  If the score `s` is greater or equal than `score_threshold`  then add `b` to `D` else go to step 8.
+7.  For each input box `b_i` from `B` and the corresponding score `s_i`, set `s_i = s_i * func(IOU(b_i, b))` and go to step 3.
+8.  Return `D`, a collection of the corresponding scores `S`, and the number of elements in `D`.
+
+Here `func(iou) = 1 if iou <= iou_threshold else 0` when `soft_nms_sigma == 0`, else `func(iou) = exp(-0.5 * iou * iou / soft_nms_sigma) if iou <= iou_threshold else 0`.
```

Comment:
I think it's still not completely correct.
I don't see `0.5` coefficient in the original formula from the paper as well as on the page 4 of the original paper right after the formula is introduced you may see that `This update rule is applied in each iteration and scores of
all remaining detection boxes are updated.` which means that we don't need to do any threshold comparison there, but only adjust the score value according to the formula.
Please correct me if I'm wrong in my suggestion.
```suggestion
Here `func(iou) = 1 if iou <= iou_threshold else 0` when `soft_nms_sigma == 0`, else `func(iou) = exp(-iou * iou / soft_nms_sigma)`.
```

Reply:
In the TF sources, the number `2 * soft_nms_sigma` is used instead of `soft_nms_sigma` of the article (see https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/image/non_max_suppression_op.cc, lines 187-196).

Reply:
Also, TF uses a  threshold comparison for all values of `soft_nms_sigma` including positive: see lines 192-196 and 221 of the file  https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/image/non_max_suppression_op.cc.